packages/electron-updater/src/NsisUpdater.ts

```diff
@@ -58,14 +59,79 @@ export class NsisUpdater extends AppUpdater {
       throw e
     }
 
-    if (logger != null) {
-      logger.info(`New version ${this.versionInfo!.version} has been downloaded to ${tempFile}`)
+    let hasValidSignature = await this.spawnVerifySignature(tempFile)
+    if (hasValidSignature) {
+      if (logger != null) {
+        logger.info(`New version ${this.versionInfo!.version} has been downloaded to ${tempFile}`)
+      }
+
+      this.setupPath = tempFile
+      this.addQuitHandler()
+      this.emit("update-downloaded", this.versionInfo)
+      return tempFile
+    } else {
+      if (logger != null) {
+        logger.error(`New version ${this.versionInfo!.version} is not signed by the application owner`)
+      }
+      return await remove(tempDir)
+    }
+  }
+
+  // $certificateInfo = (Get-AuthenticodeSignature 'xxx\yyy.exe'
+  // | where {$_.Status.Equals([System.Management.Automation.SignatureStatus]::Valid) -and $_.SignerCertificate.Subject.Contains("CN=circuitdev.siemens.com")})
+  // | Out-String ; if ($certificateInfo) { exit 0 } else { exit 1 }
+  private async spawnVerifySignature(tempUpdateFile: string) : Promise<boolean> {
+    const isWin = Platform.WINDOWS === Platform.fromString(process.platform)
```

Comment:
Platform not available in the electron-updater to reduce deps. Please use process.platform === "win32"